### PR TITLE
Reduce DafnyLanguageServer dependencies

### PR DESCRIPTION
**Pull request by keyboardDrummer**
### Changes
- Let DafnyLanguageServer depend on DafnyCore instead of DafnyPipeline, so there's no requirement to have javac installed to compile DafnyLanguageServer. This enables us to remove the javac check from the Dafny VSCode extension that is done for M1 systems.
- Let DafnyServer depend on DafnyLanguageServer instead of the other way around. Some files were moved to LanguageServer to accomodate.
- Resolve some compiler warnings

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Source/DafnyLanguageServer/DafnyLanguageServer.csproj [hunk 1]
```diff
@@ -37,9 +42,7 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\DafnyDriver\DafnyDriver.csproj" />
-    <ProjectReference Include="..\DafnyServer\DafnyServer.csproj" />
-    <ProjectReference Include="..\DafnyPipeline\DafnyPipeline.csproj" />
+    <ProjectReference Include="..\DafnyCore\DafnyCore.csproj" />
```
Comment (MikaelMayer) [inline Source/DafnyLanguageServer/DafnyLanguageServer.csproj:45]: Oh so you removed the reference to the driver as well ! Neat !

**Review by MikaelMayer — APPROVED** [review]
Not having to depend on java for the DafnyLanguageServer is a great relief, thanks !

File: Source/DafnyLanguageServer.Test/Util/Stringify.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
+#nullable enable
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyLanguageServer.Test/Util/Stringify.cs:1]: This class requires nullable to be enabled, which is not the case in the project it moved to, but was in the project it was in, so I've customised it just for this file.

File: Source/DafnyPipeline.Test/IntraMethodVerificationStability.cs [hunk 3]
```diff
@@ -165,46 +165,46 @@ class C {
     }
 
     [Fact]
-    public async void EqualProverLogWhenReorderingProgram() {
+    public async Task EqualProverLogWhenReorderingProgram() {
```
[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/DafnyPipeline.Test/IntraMethodVerificationStability.cs:168]: Seems like these were already here. Apparently we don't have warnings as error on everywhere. I'll look into it. In any case I'll try to resolve these.

**Review by MikaelMayer — APPROVED** [review]

**Pull request merged**